### PR TITLE
build: bump toolchain to nightly-2024-03-12

### PR DESCRIPTION
**Pull request by xxchan**
Signed-off-by: Richard Chien <stdrc@outlook.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
There have been quite some changes, so I want to merge now to reduce burden. 

Also since we've code freezed for 1.8, so it's a good timing now.

~~blocker https://github.com/rust-lang/rust/issues/122618~~ (workarounded)

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by MrCroxx** [comment]
Blocked by https://github.com/risingwavelabs/jsonbb/issues/3

**Comment by MrCroxx** [comment]
All tests pass now. Only fmt/clippy issues.

**Comment by MrCroxx** [comment]
https://github.com/rust-lang/rust/issues/111137

This error raises after merge main.

```
error: could not compile `curve25519-dalek` (lib) due to 5 previous errors
```

```
error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
  --> /home/mrcroxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.1/src/backend/vector/ifma/field.rs:26:5
   |
26 |     _mm256_madd52lo_epu64(z.into(), x.into(), y.into()).into()
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
   = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
   = note: this compiler was built on 2024-04-10; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
  --> /home/mrcroxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.1/src/backend/vector/ifma/field.rs:25:9
   |
25 |     use core::arch::x86_64::_mm256_madd52lo_epu64;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
   = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
   = note: this compiler was built on 2024-04-10; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
  --> /home/mrcroxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.1/src/backend/vector/ifma/field.rs:34:5
   |
34 |     _mm256_madd52hi_epu64(z.into(), x.into(), y.into()).into()
   |     ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
   = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
   = note: this compiler was built on 2024-04-10; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature 'stdarch_x86_avx512'
  --> /home/mrcroxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.1/src/backend/vector/ifma/field.rs:33:9
   |
33 |     use core::arch::x86_64::_mm256_madd52hi_epu64;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #111137 <https://github.com/rust-lang/rust/issues/111137> for more information
   = help: add `#![feature(stdarch_x86_avx512)]` to the crate attributes to enable
   = note: this compiler was built on 2024-04-10; consider upgrading it if it is out of date

error[E0635]: unknown feature `stdsimd`
  --> /home/mrcroxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/curve25519-dalek-4.1.1/src/lib.rs:13:70
   |
13 | #![cfg_attr(all(curve25519_dalek_backend = "simd", nightly), feature(stdsimd))]
   |                                                                      ^^^^^^^
```

**Comment by xxchan** [comment]
Maybe the change is lost 🤔
image

**Comment by xxchan** [comment]
@stdrc @wenym1 Please take a look at these commits. Thanks!
image

File: src/stream/src/common/log_store_impl/kv_log_store/serde.rs
Comment (wenym1) [inline src/stream/src/common/log_store_impl/kv_log_store/serde.rs:548]: Use `StateStoreReadIter`?

**Review by wenym1 — COMMENTED** [review]
Storage part LGTM!

File: src/storage/hummock_sdk/src/table_watermark.rs
Comment (stdrc) [inline src/storage/hummock_sdk/src/table_watermark.rs]: This is very suspicious. `upper_bound` "returns a Cursor pointing at the last gap below the given bound," which means if we have `[1, 2, 3, 4]`, `upper_bound(Included(3))` will return the gap between `3` and `4`, then `peek_next` will return `4`, however previously we would get `3`.

File: src/stream/src/executor/over_window/over_partition.rs
Comment (stdrc) [inline src/stream/src/executor/over_window/over_partition.rs:1]: The cursor API calls look strange, let me do some changes.

Comment (stdrc) [reply src/stream/src/executor/over_window/over_partition.rs:1]: Fixed

**Review by BugenZhao — APPROVED** [review]

**Comment by xxchan** [comment]
https://github.com/madsim-rs/madsim/issues/205

**Comment by wangrunji0408** [comment]
> [madsim-rs/madsim#205](https://github.com/madsim-rs/madsim/issues/205)

May have been fixed in the latest version (by https://github.com/madsim-rs/madsim/pull/202). I have updated madsim. Let's see if it works.

**Comment by xxchan** [comment]
:eyes: 
```
running 1 test
seed = 5
Configuration {
    config_path: Temp(
        "/tmp/.tmpKfpUSz",
    ),
    frontend_nodes: 2,
    compute_nodes: 0,
    meta_nodes: 3,
    compactor_nodes: 2,
    compute_node_cores: 2,
    etcd_timeout_rate: 0.0,
    etcd_data_path: None,
    per_session_queries: [],
}
test batch::test_serving_cluster_availability ... FAILED

failures:

failures:
    batch::test_serving_cluster_availability

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 110 filtered out; finished in 6.92s

[0m
[35;1m--- [0m[35;1mSTDERR:              [0m[35;1mrisingwave_simulation::integration_tests[0m [36mbatch[0m[36m::[0m[34;1mtest_serving_cluster_availability[0m[35;1m ---[0m
thread '<unnamed>' panicked at src/tests/simulation/tests/integration_tests/batch/mod.rs:138:31:
called `Result::unwrap_err()` on an `Ok` value: "0\n1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16\n17\n18\n19\n20\n21\n22\n23\n24\n25\n26\n27\n28\n29\n30\n31\n32\n33\n34\n35\n36\n37\n38\n39\n40\n41\n42\n43\n44\n45\n46\n47\n48\n49\n50\n51\n52\n53\n54\n55\n56\n57\n58\n59\n60\n61\n62\n63\n64\n65\n66\n67\n68\n69\n70\n71\n72\n73\n74\n75\n76\n77\n78\n79\n80\n81\n82\n83\n84\n85\n86\n87\n88\n89\n90\n91\n92\n93\n94\n95\n96\n97\n98\n99\n100\n101\n102\n103\n104\n105\n106\n107\n108\n109\n110\n111\n112\n113\n114\n115\n116\n117\n118\n119\n120\n121\n122\n123\n124\n125\n126\n127\n128\n129\n130\n131\n132\n133\n134\n135\n136\n137\n138\n139\n140\n141\n142\n143\n144\n145\n146\n147\n148\n149\n150\n151\n152\n153\n154\n155\n156\n157\n158\n159\n160\n161\n162\n163\n164\n165\n166\n167\n168\n169\n170\n171\n172\n173\n174\n175\n176\n177\n178\n179\n180\n181\n182\n183\n184\n185\n186\n187\n188\n189\n190\n191\n192\n193\n194\n195\n196\n197\n198\n199\n200\n201\n202\n203\n204\n205\n206\n207\n208\n209\n210\n211\n212\n213\n214\n215\n216\n217\n218\n219\n220\n221\n222\n223\n224\n225\n226\n227\n228\n229\n230\n231\n232\n233\n234\n235\n236\n237\n238\n239\n240\n241\n242\n243\n244\n245\n246\n247\n248\n249\n250\n251\n252\n253\n254\n255\n256\n257\n258\n259\n260\n261\n262\n263\n264\n265\n266\n267\n268\n269\n270\n271\n272\n273\n274\n275\n276\n277\n278\n279\n280\n281\n282\n283\n284\n285\n286\n287\n288\n289\n290\n291\n292\n293\n294\n295\n296\n297\n298\n299\n300\n301\n302\n303\n304\n305\n306\n307\n308\n309\n310\n311\n312\n313\n314\n315\n316\n317\n318\n319\n320\n321\n322\n323\n324\n325\n326\n327\n328\n329\n330\n331\n332\n333\n334\n335\n336\n337\n338\n339\n340\n341\n342\n343\n344\n345\n346\n347\n348\n349\n350\n351\n352\n353\n354\n355\n356\n357\n358\n359\n360\n361\n362\n363\n364\n365\n366\n367\n368\n369\n370\n371\n372\n373\n374\n375\n376\n377\n378\n379\n380\n381\n382\n383\n384\n385\n386\n387\n388\n389\n390\n391\n392\n393\n394\n395\n396\n397\n398\n399\n400\n401\n402\n403\n404\n405\n406\n407\n408\n409\n410\n411\n412\n413\n414\n415\n416\n417\n418\n419\n420\n421\n422\n423\n424\n425\n426\n427\n428\n429\n430\n431\n432\n433\n434\n435\n436\n437\n438\n439\n440\n441\n442\n443\n444\n445\n446\n447\n448\n449\n450\n451\n452\n453\n454\n455\n456\n457\n458\n459\n460\n461\n462\n463\n464\n465\n466\n467\n468\n469\n470\n471\n472\n473\n474\n475\n476\n477\n478\n479\n480\n481\n482\n483\n484\n485\n486\n487\n488\n489\n490\n491\n492\n493\n494\n495\n496\n497\n498\n499\n500\n501\n502\n503\n504\n505\n506\n507\n508\n509\n510\n511\n512\n513\n514\n515\n516\n517\n518\n519\n520\n521\n522\n523\n524\n525\n526\n527\n528\n529\n530\n531\n532\n533\n534\n535\n536\n537\n538\n539\n540\n541\n542\n543\n544\n545\n546\n547\n548\n549\n550\n551\n552\n553\n554\n555\n556\n557\n558\n559\n560\n561\n562\n563\n564\n565\n566\n567\n568\n569\n570\n571\n572\n573\n574\n575\n576\n577\n578\n579\n580\n581\n582\n583\n584\n585\n586\n587\n588\n589\n590\n591\n592\n593\n594\n595\n596\n597\n598\n599\n600\n601\n602\n603\n604\n605\n606\n607\n608\n609\n610\n611\n612\n613\n614\n615\n616\n617\n618\n619\n620\n621\n622\n623\n624\n625\n626\n627\n628\n629\n630\n631\n632\n633\n634\n635\n636\n637\n638\n639\n640\n641\n642\n643\n644\n645\n646\n647\n648\n649\n650\n651\n652\n653\n654\n655\n656\n657\n658\n659\n660\n661\n662\n663\n664\n665\n666\n667\n668\n669\n670\n671\n672\n673\n674\n675\n676\n677\n678\n679\n680\n681\n682\n683\n684\n685\n686\n687\n688\n689\n690\n691\n692\n693\n694\n695\n696\n697\n698\n699\n700\n701\n702\n703\n704\n705\n706\n707\n708\n709\n710\n711\n712\n713\n714\n715\n716\n717\n718\n719\n720\n721\n722\n723\n724\n725\n726\n727\n728\n729\n730\n731\n732\n733\n734\n735\n736\n737\n738\n739\n740\n741\n742\n743\n744\n745\n746\n747\n748\n749\n750\n751\n752\n753\n754\n755\n756\n757\n758\n759\n760\n761\n762\n763\n764\n765\n766\n767\n768\n769\n770\n771\n772\n773\n774\n775\n776\n777\n778\n779\n780\n781\n782\n783\n784\n785\n786\n787\n788\n789\n790\n791\n792\n793\n794\n795\n796\n797\n798\n799\n800\n801\n802\n803\n804\n805\n806\n807\n808\n809\n810\n811\n812\n813\n814\n815\n816\n817\n818\n819\n820\n821\n822\n823\n824\n825\n826\n827\n828\n829\n830\n831\n832\n833\n834\n835\n836\n837\n838\n839\n840\n841\n842\n843\n844\n845\n846\n847\n848\n849\n850\n851\n852\n853\n854\n855\n856\n857\n858\n859\n860\n861\n862\n863\n864\n865\n866\n867\n868\n869\n870\n871\n872\n873\n874\n875\n876\n877\n878\n879\n880\n881\n882\n883\n884\n885\n886\n887\n888\n889\n890\n891\n892\n893\n894\n895\n896\n897\n898\n899\n900\n901\n902\n903\n904\n905\n906\n907\n908\n909\n910\n911\n912\n913\n914\n915\n916\n917\n918\n919\n920\n921\n922\n923\n924\n925\n926\n927\n928\n929\n930\n931\n932\n933\n934\n935\n936\n937\n938\n939\n940\n941\n942\n943\n944\n945\n946\n947\n948\n949\n950\n951\n952\n953\n954\n955\n956\n957\n958\n959\n960\n961\n962\n963\n964\n965\n966\n967\n968\n969\n970\n971\n972\n973\n974\n975\n976\n977\n978\n979\n980\n981\n982\n983\n984\n985\n986\n987\n988\n989\n990\n991\n992\n993\n994\n995\n996\n997\n998\n999"
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/core/src/result.rs:1654:5
   3: core::result::Result<T,E>::unwrap_err
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/core/src/result.rs:1171:22
   4: integration_tests::batch::test_serving_cluster_availability::{{closure}}::{{closure}}
             at ./tests/integration_tests/batch/mod.rs:138:5
   5: madsim::sim::task::Executor::block_on::{{closure}}::{{closure}}
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:230:32
   6: async_task::raw::RawTask<F,T,S,M>::run
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.4.0/src/raw.rs:563:17
   7: async_task::runnable::Runnable<M>::run
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.4.0/src/runnable.rs:782:18
   8: madsim::sim::task::Executor::run_all_ready::{{closure}}
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:271:59
   9: std::panicking::try::do_call
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/std/src/panicking.rs:552:40
  10: std::panicking::try
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/std/src/panicking.rs:516:19
  11: std::panic::catch_unwind
             at /rustc/a07f3eb43acc5df851e15176c7081a900a30a4d7/library/std/src/panic.rs:149:14
  12: madsim::sim::task::Executor::run_all_ready
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:271:17
  13: madsim::sim::task::Executor::block_on
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:238:13
  14: madsim::sim::runtime::Runtime::block_on
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/runtime/mod.rs:126:9
  15: madsim::sim::runtime::builder::Builder::run::{{closure}}::{{closure}}::{{closure}}
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/runtime/builder.rs:128:35
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
context: node=0 "main", task=0 (spawned at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/runtime/mod.rs:126:19)
note: run with `MADSIM_TEST_SEED=5` environment variable to reproduce this error
```

**Comment by xxchan** [comment]
🤡
image

**Comment by wangrunji0408** [comment]
> 🤡 image

cc @bakjos Could you help investigate this error on the latest nightly toolchain? I have no idea where the illegal instruction is. Thanks!

**Comment by bakjos** [comment]
@wangrunji0408 I guess the issue is due to some changes done in the `nightly-2024-02-05+` although the compilation fixes were done in https://github.com/dalek-cryptography/curve25519-dalek/pull/619,  there are still some issues with the unstable `simd` feature, I think this can be fixed by forcing a different [backend](https://github.com/dalek-cryptography/curve25519-dalek/tree/main/curve25519-dalek#manual-backend-override) using 

```bash
RUSTFLAGS='--cfg curve25519_dalek_backend="serial"'
```

@xxchan, @wangrunji0408 Do you want me to push this change, to see if it fixes it?

**Comment by xxchan** [comment]
@bakjos Thanks for the investigation. I think you cannot push directly to this branch. If you are willing to help, you can create another PR against this PR’s branch and trigger CI there.

**Comment by bakjos** [comment]
@xxchan It turns out the issue is because something made between nightly 2024-03-12 and nightly 2024-03-13, the nightly 2024-03-12 compiles the code [successfully](https://buildkite.com/risingwavelabs/pull-request/builds/48845) but the 2024-03-13 [fails](https://buildkite.com/risingwavelabs/pull-request/builds/48844)  I haven't been able to figure it out what's making it fail. This is the full change list between both versions https://github.com/rust-lang/rust/compare/4a0cc881dcc4d800f10672747f61a94377ff6662...a165f1f65015b1bd4afd2ec50700aaacf2e0c485

**Comment by xxchan** [comment]
@bakjos Thanks a lot for your investigation. BTW just FYI there's a tool for finding rustc's regression https://github.com/rust-lang/cargo-bisect-rustc/

**Comment by bakjos** [comment]
@xxchan btw this error doesn't happen on osx, runs without issues using the 2024-04-26 nightly

**Comment by hzxa21** [comment]
Looking forward for the toolchain bump. I met another stale toolchain issue similar to https://github.com/zowens/crc32c/issues/58 related to arm when bumping opendal version to 0.46. In other words, without this PR, we cannot upgrade opendal version in storage.

**Comment by xxchan** [comment]
> In other words, without this PR, we cannot upgrade opendal version in storage.

This is a little unfortunate, but the situation is not that bad... 

For bumping opendal, I don't think the newly added `crc32c@0.6.5` is needed by us: https://github.com/apache/opendal/pull/4533. So a quick workaround is: we can use a patched opendal with this feature removed, or simply use `crc32c@0.6.4`.

For bumping toolchain, there are several ways to proceed:
1. Try to reproduce the core dump, and report to either deno or rust.
2. Let's trust rust, and YOLO: just disable deno UDF for now, and move on.

I don't have the energy to push forward this now. I appreciate it if anyone can take over the work.

**Comment by xxchan** [comment]
Try longevity https://buildkite.com/risingwave-test/longevity-test/builds/1394

**Comment by bakjos** [comment]
@xxchan pushed this one fixing all the clippy warnings https://github.com/risingwavelabs/risingwave/pull/16754

**Comment by xxchan** [comment]
Oops. It seems we do this chore work duplicately. 🤪

**Comment by bakjos** [comment]
:) No worries, I will close mine, seems like it's finally close to being done

**Comment by xxchan** [comment]
```
[35;1m--- [0m[35;1mSTDERR:              [0m[35;1mrisingwave_simulation::integration_tests[0m [36mbatch[0m[36m::[0m[34;1mtest_serving_cluster_availability[0m[35;1m ---[0m
thread '<unnamed>' panicked at src/tests/simulation/tests/integration_tests/batch/mod.rs:138:31:
called `Result::unwrap_err()` on an `Ok` value: "0\n1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16\n17\n18\n19\n20\n21\n22\n23\n24\n25\n26\n27\n28\n29\n30\n31\n32\n33\n34\n35\n36\n37\n38\n39\n40\n41\n42\n43\n44\n45\n46\n47\n48\n49\n50\n51\n52\n53\n54\n55\n56\n57\n58\n59\n60\n61\n62\n63\n64\n65\n66\n67\n68\n69\n70\n71\n72\n73\n74\n75\n76\n77\n78\n79\n80\n81\n82\n83\n84\n85\n86\n87\n88\n89\n90\n91\n92\n93\n94\n95\n96\n97\n98\n99\n100\n101\n102\n103\n104\n105\n106\n107\n108\n109\n110\n111\n112\n113\n114\n115\n116\n117\n118\n119\n120\n121\n122\n123\n124\n125\n126\n127\n128\n129\n130\n131\n132\n133\n134\n135\n136\n137\n138\n139\n140\n141\n142\n143\n144\n145\n146\n147\n148\n149\n150\n151\n152\n153\n154\n155\n156\n157\n158\n159\n160\n161\n162\n163\n164\n165\n166\n167\n168\n169\n170\n171\n172\n173\n174\n175\n176\n177\n178\n179\n180\n181\n182\n183\n184\n185\n186\n187\n188\n189\n190\n191\n192\n193\n194\n195\n196\n197\n198\n199\n200\n201\n202\n203\n204\n205\n206\n207\n208\n209\n210\n211\n212\n213\n214\n215\n216\n217\n218\n219\n220\n221\n222\n223\n224\n225\n226\n227\n228\n229\n230\n231\n232\n233\n234\n235\n236\n237\n238\n239\n240\n241\n242\n243\n244\n245\n246\n247\n248\n249\n250\n251\n252\n253\n254\n255\n256\n257\n258\n259\n260\n261\n262\n263\n264\n265\n266\n267\n268\n269\n270\n271\n272\n273\n274\n275\n276\n277\n278\n279\n280\n281\n282\n283\n284\n285\n286\n287\n288\n289\n290\n291\n292\n293\n294\n295\n296\n297\n298\n299\n300\n301\n302\n303\n304\n305\n306\n307\n308\n309\n310\n311\n312\n313\n314\n315\n316\n317\n318\n319\n320\n321\n322\n323\n324\n325\n326\n327\n328\n329\n330\n331\n332\n333\n334\n335\n336\n337\n338\n339\n340\n341\n342\n343\n344\n345\n346\n347\n348\n349\n350\n351\n352\n353\n354\n355\n356\n357\n358\n359\n360\n361\n362\n363\n364\n365\n366\n367\n368\n369\n370\n371\n372\n373\n374\n375\n376\n377\n378\n379\n380\n381\n382\n383\n384\n385\n386\n387\n388\n389\n390\n391\n392\n393\n394\n395\n396\n397\n398\n399\n400\n401\n402\n403\n404\n405\n406\n407\n408\n409\n410\n411\n412\n413\n414\n415\n416\n417\n418\n419\n420\n421\n422\n423\n424\n425\n426\n427\n428\n429\n430\n431\n432\n433\n434\n435\n436\n437\n438\n439\n440\n441\n442\n443\n444\n445\n446\n447\n448\n449\n450\n451\n452\n453\n454\n455\n456\n457\n458\n459\n460\n461\n462\n463\n464\n465\n466\n467\n468\n469\n470\n471\n472\n473\n474\n475\n476\n477\n478\n479\n480\n481\n482\n483\n484\n485\n486\n487\n488\n489\n490\n491\n492\n493\n494\n495\n496\n497\n498\n499\n500\n501\n502\n503\n504\n505\n506\n507\n508\n509\n510\n511\n512\n513\n514\n515\n516\n517\n518\n519\n520\n521\n522\n523\n524\n525\n526\n527\n528\n529\n530\n531\n532\n533\n534\n535\n536\n537\n538\n539\n540\n541\n542\n543\n544\n545\n546\n547\n548\n549\n550\n551\n552\n553\n554\n555\n556\n557\n558\n559\n560\n561\n562\n563\n564\n565\n566\n567\n568\n569\n570\n571\n572\n573\n574\n575\n576\n577\n578\n579\n580\n581\n582\n583\n584\n585\n586\n587\n588\n589\n590\n591\n592\n593\n594\n595\n596\n597\n598\n599\n600\n601\n602\n603\n604\n605\n606\n607\n608\n609\n610\n611\n612\n613\n614\n615\n616\n617\n618\n619\n620\n621\n622\n623\n624\n625\n626\n627\n628\n629\n630\n631\n632\n633\n634\n635\n636\n637\n638\n639\n640\n641\n642\n643\n644\n645\n646\n647\n648\n649\n650\n651\n652\n653\n654\n655\n656\n657\n658\n659\n660\n661\n662\n663\n664\n665\n666\n667\n668\n669\n670\n671\n672\n673\n674\n675\n676\n677\n678\n679\n680\n681\n682\n683\n684\n685\n686\n687\n688\n689\n690\n691\n692\n693\n694\n695\n696\n697\n698\n699\n700\n701\n702\n703\n704\n705\n706\n707\n708\n709\n710\n711\n712\n713\n714\n715\n716\n717\n718\n719\n720\n721\n722\n723\n724\n725\n726\n727\n728\n729\n730\n731\n732\n733\n734\n735\n736\n737\n738\n739\n740\n741\n742\n743\n744\n745\n746\n747\n748\n749\n750\n751\n752\n753\n754\n755\n756\n757\n758\n759\n760\n761\n762\n763\n764\n765\n766\n767\n768\n769\n770\n771\n772\n773\n774\n775\n776\n777\n778\n779\n780\n781\n782\n783\n784\n785\n786\n787\n788\n789\n790\n791\n792\n793\n794\n795\n796\n797\n798\n799\n800\n801\n802\n803\n804\n805\n806\n807\n808\n809\n810\n811\n812\n813\n814\n815\n816\n817\n818\n819\n820\n821\n822\n823\n824\n825\n826\n827\n828\n829\n830\n831\n832\n833\n834\n835\n836\n837\n838\n839\n840\n841\n842\n843\n844\n845\n846\n847\n848\n849\n850\n851\n852\n853\n854\n855\n856\n857\n858\n859\n860\n861\n862\n863\n864\n865\n866\n867\n868\n869\n870\n871\n872\n873\n874\n875\n876\n877\n878\n879\n880\n881\n882\n883\n884\n885\n886\n887\n888\n889\n890\n891\n892\n893\n894\n895\n896\n897\n898\n899\n900\n901\n902\n903\n904\n905\n906\n907\n908\n909\n910\n911\n912\n913\n914\n915\n916\n917\n918\n919\n920\n921\n922\n923\n924\n925\n926\n927\n928\n929\n930\n931\n932\n933\n934\n935\n936\n937\n938\n939\n940\n941\n942\n943\n944\n945\n946\n947\n948\n949\n950\n951\n952\n953\n954\n955\n956\n957\n958\n959\n960\n961\n962\n963\n964\n965\n966\n967\n968\n969\n970\n971\n972\n973\n974\n975\n976\n977\n978\n979\n980\n981\n982\n983\n984\n985\n986\n987\n988\n989\n990\n991\n992\n993\n994\n995\n996\n997\n998\n999"
stack backtrace:
   0: rust_begin_unwind
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/core/src/result.rs:1654:5
   3: core::result::Result<T,E>::unwrap_err
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/core/src/result.rs:1171:22
   4: integration_tests::batch::test_serving_cluster_availability::{{closure}}::{{closure}}
             at ./tests/integration_tests/batch/mod.rs:138:5
   5: madsim::sim::task::Executor::block_on::{{closure}}::{{closure}}
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:230:32
   6: async_task::raw::RawTask<F,T,S,M>::run
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.4.0/src/raw.rs:563:17
   7: async_task::runnable::Runnable<M>::run
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.4.0/src/runnable.rs:782:18
   8: madsim::sim::task::Executor::run_all_ready::{{closure}}
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:271:59
   9: std::panicking::try::do_call
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/std/src/panicking.rs:552:40
  10: std::panicking::try
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/std/src/panicking.rs:516:19
  11: std::panic::catch_unwind
             at /rustc/4a0cc881dcc4d800f10672747f61a94377ff6662/library/std/src/panic.rs:146:14
  12: madsim::sim::task::Executor::run_all_ready
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:271:17
  13: madsim::sim::task::Executor::block_on
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/task/mod.rs:238:13
  14: madsim::sim::runtime::Runtime::block_on
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/runtime/mod.rs:126:9
  15: madsim::sim::runtime::builder::Builder::run::{{closure}}::{{closure}}::{{closure}}
             at /risingwave/.cargo/registry/src/index.crates.io-6f17d22bba15001f/madsim-0.2.27/src/sim/runtime/builder.rs:128:35
```

**Comment by xxchan** [comment]
The failure can be triggered by madsim bump. https://github.com/risingwavelabs/risingwave/pull/16760

**Comment by xxchan** [comment]
> Try longevity [buildkite.com/risingwave-test/longevity-test/builds/1394](https://buildkite.com/risingwave-test/longevity-test/builds/1394)

This looks good. I think we can merge after madsim issue is resolved.

**Comment by wangrunji0408** [comment]
Seems good to go?

**Comment by xxchan** [comment]
Let's go!

**Review by MrCroxx — APPROVED** [review]

**Pull request merged**
